pkg/manifests/manifests.go

```diff
@@ -772,6 +774,94 @@ func (f *Factory) KubeStateMetricsDeployment() (*appsv1.Deployment, error) {
 	return d, nil
 }
 
+// KubeStateMetricsDenylistBoundsCheck ensures that the user is:
+// * able to enable metrics that are denied by default, and,
+// * unable to disable metrics that are enabled by default.
+func (f *Factory) KubeStateMetricsDenylistBoundsCheck(deployment *appsv1.Deployment, service *v1.Service, secret *v1.Secret) (*appsv1.Deployment, error) {
+	// Fail early if the deny-list is empty.
+	if len(f.config.ClusterMonitoringConfiguration.KubeStateMetricsConfig.MetricDenylist) == 0 {
+		return deployment, nil
+	}
+
+	// Fetch the `/metrics` port.
+	var p int32
+	for _, port := range service.Spec.Ports {
+		if port.Name == "https-main" {
+			p = port.Port
+			break
+		}
+	}
+
+	// Query the endpoint.
+	t := time.NewTimer(5 * time.Second)
+	var req *http.Request
+	var resp *http.Response
+	var err error
+	for {
+		select {
+		case <-t.C:
+			return nil, fmt.Errorf("timed out waiting to fetch kube-state-metrics /metrics endpoint: %w", err)
+		default:
+			u := &url.URL{
+				Scheme: "https",
+				Host:   fmt.Sprintf("%s.%s.svc:%d", service.GetName(), service.GetNamespace(), p),
+				Path:   "/metrics",
+			}
+			client := &http.Client{
+				Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}},
```

Comment:
Is it too complicated here to find the right TLS artifacts?
The `metrics-client-certs` secret should also have mTLS artifacts, this code wouldn't have to retrieve a bearer token for auth I think. Though there might be some fault tolerance implications when using mTLS only. @simonpasquier wdyt?

Reply:
> Is it too complicated here to find the right TLS artifacts?

I missed incorporating this, but I think that should be doable. I can use `metrics-client-certs` for the TLS certificates, and perhaps `metrics-client-ca` for the CA. I believe we'll also need to redeploy KSM if these dependent assets are rotated (same as [`metrics-server`](https://github.com/openshift/cluster-monitoring-operator/blob/bccc6b7292136f9b14dee42fc78d0b788739754d/pkg/manifests/manifests.go#L2161-L2173)).

Reply:
As a side note, I believe we should drop the following from the TLS cipher suite (for all components).

```
W0416 13:58:20.519009       1 secure_serving.go:69] Use of insecure cipher 'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256' detected.
W0416 13:58:20.519041       1 secure_serving.go:69] Use of insecure cipher TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256' detected.
```

Reply:
Mind raising a bug for us to address this?